keps/sig-node/20190129-hugepages.md

```diff
@@ -552,6 +556,18 @@ locality guarantees as a feature of QoS.  In particular, pods in the
 - E2E testing validating its usage.
 -- https://k8s-testgrid.appspot.com/sig-node-kubelet#node-kubelet-serial&include-filter-by-regex=Feature%3AHugePages
 
+## Test Plan
+
+- A test plan will consist of the following tests
+  - Unit tests
+    - Each unit test for enhancement will be implemented.
```

Comment:
Do you already have unit tests in place or will these need to be built ?
If you do have unit tests in place can you give us an idea of what they test?

Reply:
We already have unit tests.
Here are major PRs for changes and all include unittests.
It's hard to explain how unittests works just right now.
(I also need a time to take a look, I reviewed unittest of the first PR alost months ago...)
1. https://github.com/kubernetes/kubernetes/pull/84154 //merged
2. https://github.com/kubernetes/kubernetes/pull/82820 //merged
3. https://github.com/kubernetes/kubernetes/pull/84051 //derek will merge after got approval of this PR from sig-rel

Reply:
1. Container Isolation of hugepages
A unit test validates size(in a string) whether it has been transrated to right format e.g., 1024(pod spec) ->1KB(CRI)
Also, another unit test validates transformation results from v1 resource list to the list of CRI data structures.

2. Multiple sizes of hugepage support
A unit test checks node has the right amount of hugepage resources for each hugepage size as given.
Other unit test requests multiple sizes of hugeapges for a container, then checks enabled number of hugepages sizes on a container.
Also, another unit test validates whether requested medium `HugePages-<hugepagesize>` has the right format.